### PR TITLE
Project documentation (alberto.udrea0412)

### PR DESCRIPTION
**Pull request by U-alb**
### Pull Request Overview

This pull request adds my project documentation

### Testing Strategy

This pull request was tested by Alberto Udrea

### TODO or Help Wanted

This pull request isn't polished yet, I will update it constantly

### Build

- [x] Ran `npm build`.

### Author

Signed-off-by: Alberto Udrea <albertoudrea4@gmail.com>

### CODE REVIEW

File: docs/project/2024/133035.md
Comment (alexandruradovici) [inline docs/project/2024/133035.md:1]: Please use your username instead of the id.

File: docs/project/2024/133035.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Project Name
```
Comment (alexandruradovici) [inline docs/project/2024/133035.md:1]: Name your project.

File: docs/project/2024/133035.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+:::info 
+
+**Author**: Udrea Alberto-George \
+**GitHub Project Link**: [GitHub Repo](https://github.com/UPB-FILS-MA/project-UdrAlberto)
```
Comment (alexandruradovici) [inline docs/project/2024/133035.md:6]:
```suggestion
**GitHub Project Link**: https://github.com/UPB-FILS-MA/project-UdrAlberto
```

File: docs/project/2024/133035.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+-->
+
+| Device | Usage | Price |
+|--------|--------|-------|
+| [Rapspberry Pi Pico WH](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html) | The microcontroller | [56 RON](https://ardushop.ro/ro/home/2819-raspberry-pi-pico-wh.html) |
+| [BME280](https://www.bosch-sensortec.com/products/environmental-sensors/humidity-sensors-bme280/) | Temperature, pressure and humidity sensor | [5,9 EUR](https://eu.mouser.com/ProductDetail/Bosch-Sensortec/BME280) |
+| LED/OLED Display | Display | |
+| Breadboard |
+| Jumper Wires |
```
Comment (alexandruradovici) [inline docs/project/2024/133035.md:60]:
```suggestion
| Breadboard | | |
| Jumper Wires | | |
```

Comment (alexandruradovici) [reply docs/project/2024/133035.md:60]: Add the price for the breadboard and for the jumper wires and a link to them.

File: docs/project/2024/133035.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+| Push buttons | For switching info on the screen|
```
Comment (alexandruradovici) [inline docs/project/2024/133035.md:61]: Add the link and price please.

File: docs/project/2024/133035.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+```
+
+-->
+
+| Device | Usage | Price |
+|--------|--------|-------|
+| [Rapspberry Pi Pico WH](https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html) | The microcontroller | [56 RON](https://ardushop.ro/ro/home/2819-raspberry-pi-pico-wh.html) |
+| [BME280](https://www.bosch-sensortec.com/products/environmental-sensors/humidity-sensors-bme280/) | Temperature, pressure and humidity sensor | [5,9 EUR](https://eu.mouser.com/ProductDetail/Bosch-Sensortec/BME280) |
+| LED/OLED Display | Display | |
```
Comment (alexandruradovici) [inline docs/project/2024/133035.md:58]: Add the link and price please.

File: docs/project/2024/your_curs_upb_ro_id.md
Comment (alexandruradovici) [inline docs/project/2024/your_curs_upb_ro_id.md:1]: Please do not delete this file.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Comment by alexandruradovici** [comment]
I converted this to a draft as some parts are still missing.

File: docs/project/2024/alberto.udrea0412/index.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+The project builds a weather station which should provide real-time weather data on an external display and web interface
+
+## Motivation
+
+Building a Raspberry Pi Pico W weather station with embassy-rs combined my interest in embedded systems with a chance to explore networking. It allows me to learn sensor interaction, data processing, and real-time programming. Integrating a web interface adds a network layer, enabling real-time data display and opening doors for future functionalities like remote monitoring.
+
+## Architecture 
+
+Add here the schematics with the architecture of your project. Make sure to include:
```
Comment (alexandruradovici) [inline docs/project/2024/alberto.udrea0412/index.md:20]: Please dewcribe the architecture of the project.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Review by alexandruradovici — COMMENTED** [review]
Did you test the website locally?

**Comment by U-alb** [comment]
> Did you test the website locally?

Yes, and I also noticed an issue which I'm not sure how to fix: the arrows in the schematic are black, and that makes them virtually invisible if the site is set on dark mode. 

Help would be appreciated

**Comment by alexandruradovici** [comment]
Take a look at https://docusaurus.io/docs/next/markdown-features/assets and https://docusaurus.io/docs/next/markdown-features/assets. Please test the website with `npm run build` and `npm serve`.

File: docs/project/2024/alberto.udrea0412/index.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+The project builds a weather station which should provide real-time weather data on an external display and web interface
+
+## Motivation
+
+Building a Raspberry Pi Pico W weather station with embassy-rs combined my interest in embedded systems with a chance to explore networking. It allows me to learn sensor interaction, data processing, and real-time programming. Integrating a web interface adds a network layer, enabling real-time data display and opening doors for future functionalities like remote monitoring.
+
+## Architecture 
+
+![Architecture](architecture_light.svg#gh-light-mode-only)]![Architecture](architecture_dark.svg#gh-dark-mode-only)
```
Comment (U-alb) [inline docs/project/2024/alberto.udrea0412/index.md:20]: I tried adding theme-specific files here, but their visibility doesn't seems to be affected by it resulting in both images being visible at once. I used the GitHub image theming approach, should I have used something else?

File: docs/project/2024/alberto.udrea0412/index.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (U-alb) [inline docs/project/2024/alberto.udrea0412/index.md:21]: I tried adding theme-specific files here, but their visibility doesn't seems to be affected by it resulting in both images being visible at once. I used the GitHub image theming approach, should I have used something else?

**Comment by alexandruradovici** [comment]
You have to add 

```css
[data-theme='light'] img[src$='#gh-dark-mode-only'],
[data-theme='dark'] img[src$='#gh-light-mode-only'] {
  display: none;
}
```

to `src/css/custom.css`.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Please add the required css styles.

**Comment by U-alb** [comment]
> Please add the required css styles.

Unfortunately I have no access to a computer but I'll try to implement the suggestions as soon as possible. Thank you for the help!

**Comment by alexandruradovici** [comment]
I sent you a pull request with the fix, https://github.com/U-alb/upb-fils-ma.github.io/pull/1.

File: docs/project/2024/alberto.udrea0412/index.md [hunk 9]
```diff
@@ -0,0 +1,78 @@
+# Weather Station
+
```
Comment (alexandruradovici) [inline docs/project/2024/alberto.udrea0412/index.md:2]: Please add a one line description of your project.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: docs/project/2024/alberto.udrea0412/index.md (hunk 9 above)
Comment (U-alb) [reply docs/project/2024/alberto.udrea0412/index.md:2]: In the description section (i.e. shorten the existing) or make a new shorter description right below the title?

Comment (alexandruradovici) [reply docs/project/2024/alberto.udrea0412/index.md:2]: A shorter description under the title.

**Review by alexandruradovici — APPROVED** [review]

**Review by TeodoraMiu — APPROVED** [review]

**Pull request merged**